animation_nodes/nodes/gpencil/gp_layer_from_frames.py

```diff
@@ -1,36 +1,71 @@
+        self.newInput("Text", "Blend Mode", "blendMode", value = 'REGULAR')
+        self.newInput("Float", "Opacity", "opacity", value = 1, minValue = 0, maxValue = 1)
+        self.newInput("Boolean", "Use Lights", "useLights", value = True, hide = True)
         self.newInput("Color", "Tint Color", "tintColor", hide = True)
         self.newInput("Float", "Tint Factor", "tintFactor", value = 0, minValue = 0, maxValue = 1, hide = True)
         self.newInput("Float", "Stroke Thickness", "lineChange", hide = True)
         self.newInput("Integer", "Pass Index", "passIndex", value = 0, minValue = 0, hide = True)
-        self.newInput("Boolean", "Mask Layer", "maskLayer", value = False, hide = True)
+        self.newInput(VectorizedSocket("GPLayer", "useMaskLayerList",
+            ("Mask Layer", "maskLayer"), ("Mask Layers", "maskLayers")), hide = True)
+        self.newInput(VectorizedSocket("Boolean", "useInvertMaskLayerList",
+            ("Invert Mask Layer", "invertMaskLayer"), ("Invert Mask Layers", "invertMaskLayers")),
+            value = False, hide = True)
+
         self.newOutput("GPLayer", "Layer", "layer")
 
-    def execute(self, frames, layerName, blendMode, opacity, tintColor, tintFactor, lineChange, passIndex, maskLayer):
+    def execute(self, frames, layerName, blendMode, opacity, useLights, tintColor, tintFactor, lineChange,
+                passIndex, maskLayersIn, invertMaskLayers):
         if not self.useFrameList:
             frames = [frames]
+        if not self.useMaskLayerList:
+            maskLayersIn = [maskLayersIn]
 
         frameNumbers = [frame.frameNumber for frame in frames]
         if len(frameNumbers) != len(set(frameNumbers)):
             self.raiseErrorMessage("Some Frame Numbers are repeated.")
+        if layerName == "":
+            self.raiseErrorMessage("Layer name can't be empty.")
         if blendMode not in ['REGULAR', 'OVERLAY', 'ADD', 'SUBTRACT', 'MULTIPLY', 'DIVIDE']:
             self.raiseErrorMessage("The blend mode is invalid. \n\nPossible values for 'Blend Mode' are: 'REGULAR', 'OVERLAY', 'ADD', 'SUBTRACT', 'MULTIPLY', 'DIVIDE'")
-        return GPLayer(layerName, frames, blendMode, opacity, tintColor, tintFactor, lineChange, passIndex, maskLayer)
+
+        maskLayers = []
+        if len(maskLayersIn) > 0:
+            invertMaskLayers = VirtualBooleanList.create(invertMaskLayers, False)
+            for i, maskLayer in enumerate(maskLayersIn):
+                if self.isEmptyLayer(maskLayer): continue
+                maskLayerName = maskLayer.layerName
+                if maskLayerName != "" and maskLayerName != layerName:
+                    maskLayer.invertAsMask = invertMaskLayers[i]
+                    maskLayers.append(maskLayer)
+
+        return GPLayer(layerName, frames, blendMode, opacity, useLights, tintColor, tintFactor, lineChange,
+                       passIndex, False, maskLayers)
+
+    def isEmptyLayer(self, layer):
```

Comment:
You can move this to the GPLayer class as a class method, notice that it is also used in `set_gp_layer_attributes.py` Also, you can do something like this:
```
def isEmptyLayer(self):
    for frame in self.frames:
        if len(frame.strokes) > 0:
            return False
    return True
```

Reply:
Okay.

Reply:
Done. BTW why class method not as regular method?

Reply:
@3DSinghVFX My apologies, I meant a normal method.

Reply:
Okay, I'll change that.

Should I removed this `if len(maskLayersIn) > 0`?. I added for the performance:
https://github.com/JacquesLucke/animation_nodes/blob/5c8005435cd226c5f3783899b7137bd0f4795fcf/animation_nodes/nodes/gpencil/gp_layer_from_frames.py#L50-L58

Reply:
@3DSinghVFX  It shouldn't make much of a difference. It is up to you.

Reply:
Okay. I'll keep that.